packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -57,20 +57,50 @@ public function setBuildVersion(int $buildVersion): void
      * @param int $domainId
      * @param string $index
      * @return string
+     * @deprecated Getting index name without a version using this method is deprecated since SSFW 7.3, use getVersionedIndexName or getAliasName instead
      */
     public function getIndexName(int $domainId, string $index): string
+    {
+        @trigger_error(
+            sprintf('Getting index name without a version using method "%s" is deprecated since SSFW 7.3, use %s or %s instead', __METHOD__, 'getVersionedIndexName', 'getAliasName'),
+            E_USER_DEPRECATED
+        );
+        return $this->indexPrefix . $index . $domainId;
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     * @return string
+     */
+    public function getVersionedIndexName(int $domainId, string $index): string
     {
         return $this->indexPrefix . $this->buildVersion . $index . $domainId;
     }
 
+    /**
+     * @param string $aliasName
+     * @return string
+     */
+    public function getCurrentIndexName(string $aliasName): string
+    {
+        $indexNames = $this->getExistingIndexNamesForAlias($aliasName);
+
+        if (count($indexNames) > 1) {
+            throw new ElasticsearchStructureException('There is more than one index set as alias. Please delete all non current ones.');
+        } else {
+            return reset($indexNames);
+        }
+    }
+
     /**
      * @param int $domainId
      * @param string $index
      * @return string
      */
     public function getAliasName(int $domainId, string $index): string
     {
-        return $this->indexPrefix . $index . $domainId;
+        return $this->indexPrefix . $index . $domainId . 'alias';
```

Comment:
I would rather put a separator there: `. '_alias'`

Reply:
I removed the suffix